### PR TITLE
CHANGELOG: fix punctuation errors in 1.21

### PR DESCRIPTION
**Pull request by wangyx1992**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR fixes some punctuation errors in CHANGELOG.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Review by soniasingla — APPROVED** [review]
/lgtm
/approve

**Comment by wangyx1992** [comment]
Hi, @hasheddan @wilsonehusin. This PR fixes some punctuation errors in 1.21, waiting for your review and approval with pleasure.

File: CHANGELOG/CHANGELOG-1.21.md
Comment (wilsonehusin) [inline CHANGELOG/CHANGELOG-1.21.md:1374]: `-gce-zones` is correctly written

```
❯ docker run --rm -it k8s.gcr.io/conformance:v1.21.0 e2e.test -h
...
  -gce-zone string
    	GCE zone being used, if applicable
  -gce-zones value
    	The set of zones to use in a multi-zone test instead of querying the cloud provider.
...
```

File: CHANGELOG/CHANGELOG-1.21.md
Comment (wilsonehusin) [inline CHANGELOG/CHANGELOG-1.21.md:1327]: interesting, this seems to be a bug in the way we parse content. let's preserve the original message:

```suggestion
- New admission controller "denyserviceexternalips" is available.  Clusters which do not *need* the Service "externalIPs" feature should enable this controller and be more secure. ([#97395](https://github.com/kubernetes/kubernetes/pull/97395), [@thockin](https://github.com/thockin)) [SIG API Machinery]
```

File: CHANGELOG/CHANGELOG-1.21.md
Comment (wilsonehusin) [inline CHANGELOG/CHANGELOG-1.21.md:1028]: same here, let's preserve the original intent of the message:

```suggestion
- Stdin is now only passed to client-go exec credential plugins when it is detected to be an interactive terminal. Previously, it was passed to client-go exec plugins when **stdout** was detected to be an interactive terminal. ([#99654](https://github.com/kubernetes/kubernetes/pull/99654), [@ankeesler](https://github.com/ankeesler)) [SIG API Machinery and Auth]
```

**Review by wilsonehusin — CHANGES_REQUESTED** [review]
good catch on the bug! can you also fix line 1356 where there is an instance of `*need*` being converted to `*need-`?

everything else is fine, thanks!

**Comment by wangyx1992** [comment]
> good catch on the bug! can you also fix line 1356 where there is an instance of `*need*` being converted to `*need-`?
> 
> everything else is fine, thanks!

Hi, @wilsonehusin. The above have been updated already.
In line 1356, `Kubeadm: fix bad token placeholder text in "config print *-defaults --help"`. According to [https://github.com/kubernetes/kubernetes/pull/98839](url),  `*` expresses the parameter passed to `newCmdConfigPrintActionDefaults` . So this does not need to modify.

**Comment by wilsonehusin** [comment]
ah I see, thanks for double checking! :smile: turns out i have and older copy locally when searching.

/ok-to-test
/approve

**Pull request merged**
